src/qililab/qprogram/qblox_compiler.py

```diff
@@ -341,15 +342,34 @@ def _handle_wait(self, element: Wait):
         if isinstance(element.duration, Variable):
             duration = self._buses[element.bus].variable_to_register[element.duration]
             self._buses[element.bus].dynamic_durations.append(element.duration)
+            self._buses[element.bus].qpy_block_stack[-1].append_component(
+                component=QPyInstructions.Wait(wait_time=duration)
+            )
         else:
             convert = QbloxCompiler._convert_value(element)
             duration = convert(element.duration)
             self._buses[element.bus].static_duration += duration
-        self._buses[element.bus].qpy_block_stack[-1].append_component(
-            component=QPyInstructions.Wait(wait_time=duration)
-        )
+            # loop over wait instructions if static duration is longer than allowed qblox max wait time of 2**16 -4
+            if duration > INST_MAX_WAIT:
+                self._handle_long_wait(bus=element.bus, duration=duration)
+            # add the remaining wait time (or all of it if the above conditional is false)
+            self._buses[element.bus].qpy_block_stack[-1].append_component(
+                component=QPyInstructions.Wait(wait_time=duration % INST_MAX_WAIT)
+            )
         self._buses[element.bus].marked_for_sync = True
 
+    def _handle_long_wait(self, bus: str, duration: int):
+        """Wait for longer than QBLOX INST_MAX_WAIT by looping over wait instructions
+
+        Args:
+            element (Wait): wait element
+            duration (int): duration to wait in ns
+        """
+        for _ in range(duration // INST_MAX_WAIT):
+            self._buses[bus].qpy_block_stack[-1].append_component(
+                component=QPyInstructions.Wait(wait_time=INST_MAX_WAIT)
+            )
```

Comment:
Since this method does not actually handle long waits by itself (since you also need the extra lines at 356-458) I would prefer to have it directly as code inside the `_handle_wait`. It will be also be clearer since it is not a "handle", it's not added to the `_handlers` dictionary.